apps/passport-server/src/routing/pcdHttpError.ts

```diff
@@ -0,0 +1,39 @@
+import { Response } from "express";
+
+/**
+ * Throw an instance of this error in server application within an express
+ * request handler to to return a custom status code, and optionally a custom
+ * error message.
+ */
+export class PCDHTTPError extends Error {
+  public readonly code: number;
+
+  public constructor(code: number, message?: string, options?: ErrorOptions) {
+    super(message, options);
+    this.code = code;
+  }
+}
+
+/**
+ * If {@link e} is an instance of {@link PCDHTTPError}, returns with the error code
+ * and message specified in the error.
+ *
+ * If no message is provided within the {@link PCDHTTPError}, returns the default message
+ * for that http status code.
+ *
+ * If {@link e} is not an {@link PCDHTTPError}, returns with a generic 500 server error.
+ */
+export function respondWithError(
```

Comment:
I like this pattern, and how it's used in params.ts.  Seems a clean way to report things server side.

Can/should these things propagate similarly client-side?  I don't see any sign that the client distinguishes a 404 from other error types, for instance.

Reply:
The client sometimes distinguishes a 404 from a 500. Eg. `requestError`.